src/Filter/Filter.php

```diff
@@ -24,6 +24,17 @@ abstract class Filter extends BaseFilter
      */
     protected $active = false;
 
+    /**
+     * NEXT_MAJOR change $queryBuilder type for ProxyQuery.
+     *
+     * Apply the filter to the QueryBuilder instance.
+     *
+     * @param string  $alias
+     * @param string  $field
+     * @param mixed[] $value
+     */
+    abstract public function filter(ProxyQueryInterface $queryBuilder, $alias, $field, $value);
```

Comment:
This is a BC break. You MUST implement the new method, if you create a custom filter

Reply:
This is not, because the method is not new. It's in the FilterInterface.

It will be removed in the next major, but will still be used in this persistence bundle.